examples/jsm/postprocessing/OutlinePass.js

```diff
@@ -306,6 +296,8 @@ class OutlinePass extends Pass {
 
 			renderer.setClearColor( 0xffffff, 1 );
 
+			// set the selected meshes flag to recalculate selected meshes in case the selectedObjects array changed since last frame
+			this._selectedMeshesDirty = true;
```

Comment:
Instead of introducing a new flag, it might be easier to have `updateSelectionSet()` which is called once in `render()`. Then `changeVisibilityOfNonSelectedObjects()` can work with the set. After that, the set is cleared.

The set itself can then be a module scope variable, I doubt there will be any side effects.

Reply:
I did some refactoring to make things more consistent. The set is now a member like `_visibilityCache` and called `_selectionCache()` . Would these changes work for you?

Reply:
Yes, that makes sense.